include/envoy/request_info/request_info.h

```diff
@@ -213,6 +223,12 @@ class RequestInfo {
    */
   virtual Upstream::HostDescriptionConstSharedPtr upstreamHost() const PURE;
 
+  /**
+   * @param upstream_local_address sets upstream local address.
+   */
+  virtual void setUpstreamLocalAddress(
```

Comment:
This is rather unintuitive. What does upstream local or downstream local mean?

Reply:
After looking at the rest of the code, may be it might be helpful to add a line or two describing that upstream local is needed as we allow Envoy to bind to a different ip when making outbound calls from the cluster.